### PR TITLE
fix clippy issue

### PR DESCRIPTION
**Pull request by shanyp**

### CODE REVIEW

File: pageserver/src/pgdatadir_mapping.rs [hunk 1]
```diff
@@ -675,7 +675,7 @@ impl Timeline {
 
         result.add_key(CONTROLFILE_KEY);
         result.add_key(CHECKPOINT_KEY);
-        if let Ok(_) = self.get(AUX_FILES_KEY, lsn, ctx).await {
+        if (self.get(AUX_FILES_KEY, lsn, ctx).await).is_ok() {
```
Comment (arpad-m) [inline pageserver/src/pgdatadir_mapping.rs:678]:
```suggestion
        if self.get(AUX_FILES_KEY, lsn, ctx).await.is_ok() {
```

**Review by arpad-m — APPROVED** [review]
is this for nightly/beta clippy? IIRC we check clippy in CI and it doesn't break there, but we only check stable clippy. the new code is definitely better though.

**Comment by shanyp** [comment]
> is this for nightly/beta clippy? IIRC we check clippy in CI and it doesn't break there, but we only check stable clippy. the new code is definitely better though.

it was failing in the ci of current non merged PR

**Comment by arpad-m** [comment]
Ah this was about #5660. I didn't see that the PR wasn't for the main branch.

**Pull request merged**